packages/block-editor/src/components/block-toolbar/index.js

```diff
@@ -179,36 +179,26 @@ export function PrivateBlockToolbar( {
 			key={ toolbarKey }
 		>
 			<div ref={ toolbarWrapperRef } className={ innerClasses }>
-				{ ! isMultiToolbar &&
-					isLargeViewport &&
-					isDefaultEditingMode && <BlockParentSelector /> }
+				{ ! isMultiToolbar && isLargeViewport && (
+					<BlockParentSelector />
+				) }
 				{ ( shouldShowVisualToolbar || isMultiToolbar ) &&
-					( isDefaultEditingMode ||
-						( isContentOnlyEditingMode && ! hasParentPattern ) ||
-						isSynced ) && (
+					! hasParentPattern && (
```

Comment:
I left this exception in place but I believe we should remove it as well (we need to move the pattern binding to block-editor package first) cc @talldan

Reply:
Made an issue here - https://github.com/WordPress/gutenberg/issues/65486